core/src/test/scala/tech/beshu/ror/unit/boot/RorIndexTest.scala

```diff
@@ -0,0 +1,385 @@
+  with Inside with OptionValues with EitherValues
+  with MockFactory with Eventually {
+
+  implicit private val envVarsProvider: EnvVarsProvider = OsEnvVarsProvider
+  private implicit val testClock: Clock = Clock.systemUTC()
+
+  private val defaultRorIndexName: NonEmptyString = ".readonlyrest"
+  private val customRorIndexName: NonEmptyString = "custom_ror_index"
+
+  private val mainRorConfigDocumentId = "1"
+  private val testRorConfigDocumentId = "2"
+
+  "A ReadonlyREST core" should {
+    "load ROR index name" when {
+      "no index is defined in config" should {
+        "start ROR with default index name" in {
+          val resourcesPath = "/boot_tests/index_config/no_index_defined/"
+          val indexJsonContentService = mock[IndexJsonContentService]
+
+          (indexJsonContentService.sourceOf _)
+            .expects(fullIndexName(defaultRorIndexName), mainRorConfigDocumentId)
+            .once()
+            .returns(Task.now(Right(Map("settings" -> indexRorConfig.raw))))
+
+          (indexJsonContentService.sourceOf _)
+            .expects(fullIndexName(defaultRorIndexName), testRorConfigDocumentId)
+            .once()
+            .returns(Task.now(Left(IndexJsonContentService.ContentNotFound)))
+
+          val coreFactory = mock[CoreFactory]
+          mockCoreFactory(coreFactory, indexRorConfig)
+
+          val result = readonlyRestBoot(
+            coreFactory,
+            indexJsonContentService,
+            resourcesPath
+          )
+            .start()
+            .runSyncUnsafe()
+
+          result shouldBe a[Right[_, RorInstance]]
+        }
+        "save ROR config in default index" in {
+          val resourcesPath = "/boot_tests/index_config/no_index_defined/"
+          val indexJsonContentService = mock[IndexJsonContentService]
+
+          (indexJsonContentService.sourceOf _)
+            .expects(fullIndexName(defaultRorIndexName), mainRorConfigDocumentId)
+            .once()
+            .returns(Task.now(Right(Map("settings" -> indexRorConfig.raw))))
+
+          (indexJsonContentService.sourceOf _)
+            .expects(fullIndexName(defaultRorIndexName), testRorConfigDocumentId)
+            .once()
+            .returns(Task.now(Left(IndexJsonContentService.ContentNotFound)))
+
+          val coreFactory = mock[CoreFactory]
+          mockCoreFactory(coreFactory, indexRorConfig)
+
+          val result = readonlyRestBoot(
+            coreFactory,
+            indexJsonContentService,
+            resourcesPath
+          )
+            .start()
+            .runSyncUnsafe()
+
+          result shouldBe a[Right[_, RorInstance]]
+
+          mockCoreFactory(coreFactory, rorConfig)
+
+          (indexJsonContentService.saveContent _)
+            .expects(fullIndexName(defaultRorIndexName), mainRorConfigDocumentId, Map("settings" -> rorConfig.raw))
+            .once()
+            .returns(Task.now(Right(())))
+
+          val rorInstance = result.right.value
+
+          rorInstance
```

Comment:
let's assign the result to a well-named variable (eg. forceReloadingResult or sth like that)